### ISSUE TITLE
Either learning rate is logged incorrectly, or LinearWarmupCosineAnnealingLR isn't properly stepped

### ISSUE THREAD

**Issue by RylanSchaeffer** [Lightning-AI/lightning#14354]
## 🐛 Bug

I'm using an `SGD` optimizer with a PL Bolts `LinearWarmupCosineAnnealingLR` scheduler with initial learning rate = 0. However, during training, my logged learning rate never changes. Either the learning rate is incorrectly logged, or the scheduler isn't properly stepped.

![image](https://user-images.githubusercontent.com/8942987/186047829-89283546-98a9-47d5-9cb9-c738735adb95.png)

### Code

Optimizer:

```
            optimizer = torch.optim.SGD(
                self.parameters(),
                lr=self.wandb_config['learning_rate'],
                weight_decay=self.wandb_config['weight_decay'],
                momentum=0.9,
            )
```

Scheduler:

```
            scheduler = pl_bolts.optimizers.lr_scheduler.LinearWarmupCosineAnnealingLR(
                optimizer=optimizer,
                warmup_epochs=10,
                max_epochs=self.wandb_config['n_epochs'],
                warmup_start_lr=0.,
            )
```

Learning Rate Monitor:

```
lr_monitor_callback = LearningRateMonitor(
    logging_interval='step',
    log_momentum=True)
```

### Environment

```
$ python collect_env_details.py
* CUDA:
        - GPU:               None
        - available:         False
        - version:           11.3
* Lightning:
        - lightning-bolts:   0.5.0
        - pytorch-lightning: 1.7.0
        - pytorch-optimizer: 1.1.2
        - pytorch-ranger:    0.1.1
        - torch:             1.12.1+cu113
        - torch-optimizer:   0.3.0
        - torchaudio:        0.12.1+cu113
        - torchmetrics:      0.9.3
        - torchvision:       0.13.1+cu113
* Packages:
        - absl-py:           1.2.0
        - aiohttp:           3.8.1
        - aiosignal:         1.2.0
        - async-timeout:     4.0.2
        - asynctest:         0.13.0
        - attrs:             22.1.0
        - cachetools:        5.2.0
        - certifi:           2022.6.15
        - charset-normalizer: 2.1.0
        - click:             8.1.3
        - cycler:            0.11.0
        - docker-pycreds:    0.4.0
        - fonttools:         4.34.4
        - frozenlist:        1.3.1
        - fsspec:            2022.7.1
        - gitdb:             4.0.9
        - gitpython:         3.1.27
        - google-auth:       2.10.0
        - google-auth-oauthlib: 0.4.6
        - grpcio:            1.47.0
        - idna:              3.3
        - importlib-metadata: 4.12.0
        - kiwisolver:        1.4.4
        - lightning-bolts:   0.5.0
        - markdown:          3.4.1
        - markupsafe:        2.1.1
        - matplotlib:        3.5.2
        - multidict:         6.0.2
        - numpy:             1.21.6
        - oauthlib:          3.2.0
        - packaging:         21.3
        - pandas:            1.3.5
        - pathtools:         0.1.2
        - pillow:            9.2.0
        - pip:               22.2.2
        - promise:           2.3
        - protobuf:          3.19.4
        - psutil:            5.9.1
        - pyasn1:            0.4.8
        - pyasn1-modules:    0.2.8
        - pydeprecate:       0.3.2
        - pyparsing:         3.0.9
        - python-dateutil:   2.8.2
        - pytorch-lightning: 1.7.0
        - pytorch-optimizer: 1.1.2
        - pytorch-ranger:    0.1.1
        - pytz:              2022.1
        - pyyaml:            6.0
        - requests:          2.28.1
        - requests-oauthlib: 1.3.1
        - rsa:               4.9
        - scipy:             1.7.3
        - seaborn:           0.11.2
        - sentry-sdk:        1.9.2
        - setproctitle:      1.3.1
        - setuptools:        63.4.2
        - shortuuid:         1.0.9
        - six:               1.16.0
        - smmap:             5.0.0
        - tensorboard:       2.9.1
        - tensorboard-data-server: 0.6.1
        - tensorboard-plugin-wit: 1.8.1
        - torch:             1.12.1+cu113
        - torch-optimizer:   0.3.0
        - torchaudio:        0.12.1+cu113
        - torchmetrics:      0.9.3
        - torchvision:       0.13.1+cu113
        - tqdm:              4.64.0
        - typing-extensions: 4.3.0
        - urllib3:           1.26.11
        - wandb:             0.13.1
        - werkzeug:          2.2.1
        - wheel:             0.37.1
        - yarl:              1.8.1
        - zipp:              3.8.1
* System:
        - OS:                Linux
        - architecture:
                - 64bit
                - 
        - processor:         x86_64
        - python:            3.7.4
        - version:           #1 SMP Wed Aug 7 18:08:02 UTC 2019
```

**Comment by manangoel99** [comment]
I think the reason for this is explained [here](https://github.com/Lightning-AI/lightning/issues/13382#issuecomment-1164534848)

**Comment by RylanSchaeffer** [comment]
I discovered the cause of the problem: my `configure_optimizers()` method returned the learning rate scheduler in a dictionary with corresponding key `scheduler`, but the key needed to be `lr_scheduler`.
